picard/ui/metadatabox.py

```diff
@@ -585,14 +585,20 @@ def _update_tags(self, new_selection=True, drop_album_caches=False):
         top_tags = config.setting['metadatabox_top_tags']
         top_tags_set = set(top_tags)
 
+        # format_settings = {
+        #     "write_id3v23": config.setting["write_id3v23"],
+        #     "id3v23_join_with": config.setting["id3v23_join_with"],
+        # }
+        format_settings = None
```

Comment:
I don't think you can replace a call by reference object with a different object and have it passed back. So that you keep the same object, I think you need to set this to an empty dict rather than None and change the code in format_specific_metadata to something like:
```
    def format_specific_metadata(self, metadata, tag, settings={}):
        if not settings:
            config = get_config()
            settings["write_id3v23"] = config.setting["write_id3v23"]
            settings["id3v23_join_with"] = config.setting["id3v23_join_with"]
```

Reply:
I tested this in a sandbox with:
```
variable = None
def change(v):
    v = {"item": "text"}
change(variable)
print(variable)
```
and got `None` printed.

Reply:
Not sure I follow you. Actually you should not set the default parameter to an object, as this will create an object only once. You would always manipulate the same object on later calls. Instead creating the object when it is empty makes sure the object does exist only for that call.

Reply:
Try my example and see why it will pass None in every time and get settings every time.

You need to set an empty Dict object in the calling routine and add items to it in the called routine in order that the added items are passed back to the calling routine and then passed into the called routine on the second call - thus avoiding the expensive config settings calls.

Reply:
Ah, now I see what you mean. No, that is not supposed to pass anything back. That just happens because in the caller the code to set the config is commented out. I only did this for debugging, I'll update that :)

Reply:
I don't think the code should be set in the calling routing - it should just pass an empty dict object - the contents of the dict object should be set inside format_specific_mnetadata because: a) that avoids code duplication and b) f_s_m knows what settings it needs to cache.